packages/core/src/utils/typenames.ts

```diff
@@ -83,6 +84,11 @@ export const formatDocument = <T extends DocumentNode>(node: T): T => {
     });
 
     formattedDocs.set(query.__key, result);
+    if (typeof node !== 'string' && node.loc?.source.name === 'gql') {
+      const printed = print(result);
```

Comment:
I'm not really sure about this approach 🤔 I see why this happens, i.e. if both sides are stringifier, but I think the deep update and print here seems a little odd to me.